lms/djangoapps/program_enrollments/management/commands/send_program_course_nudge_email.py

```diff
@@ -0,0 +1,163 @@
+        """
+        passed_course_to_users_maps = defaultdict(list)
+        yesterday = timezone.now().date() - timedelta(days=1)
+        passed_grades = PersistentCourseGrade.objects.filter(
+            passed_timestamp__date=yesterday,
+        )
+
+        passing_user_ids = list(passed_grades.values_list('user_id', flat=True).distinct())
+        passing_users = User.objects.filter(id__in=passing_user_ids)
+        user_id_to_user_map = {user.id: user for user in passing_users}
+
+        for passing_grade in passed_grades:
+            user = user_id_to_user_map[passing_grade.user_id]
+            passed_course_to_users_maps[str(passing_grade.course_id)].append(user)
+
+        LOGGER.info(
+            '[Program Course Nudge Email] Found [%s] passing grades on [%s] date with [%s] distinct users '
+            'and [%s] distinct courses',
+            passed_grades.count(),
+            yesterday,
+            len(passing_user_ids),
+            len(passed_course_to_users_maps.keys()),
+        )
+
+        return passed_course_to_users_maps
+
+    def get_course_run_to_suggest(self, programs_progress):
+        """
+        Finds out enrollable course run from programs Generated by ProgramProgressMeter.
+        """
+        for program in programs_progress:
+            for not_started_courses in program['not_started']:
+                for course_run in not_started_courses['course_runs']:
+                    if (course_run['is_enrollable']
+                            and course_run['is_marketable']
+                            and course_run['marketing_url']
+                            and course_run['status'] == COURSE_PUBLISHED):
+                        return course_run
+        return None
+
+    def sort_programs(self, programs):
+        """
+        Sorts programs based on their revenue ranking.
+        """
+        sort_revenue_order = {
+            'MicroMasters': 1,
+            'Professional Program': 2,
+            'Professional Certificate': 3,
+            'XSeries': 4,
+            'Masters': 5,
+            'MicroBachelors': 6,
+        }
+        for program in programs:
+            program['sort_revenue_order'] = sort_revenue_order.get(program['type'], 7)
+
+        return sorted(programs, key=itemgetter('sort_revenue_order'))
+
+    def emit_event(self, user, course_run):
+        """
+         Emit the Segment event which will be used by Braze to send the email
+        """
+        failure_url = urljoin(settings.LMS_ROOT_URL, 'dashboard')
+        event_properties = {
+            'course_id': course_run['key'],
+            'username': user.username,
+            'course_url': urljoin(settings.MKTG_URLS.get('ROOT'), course_run['marketing_url']),
+            'course_title': course_run['title'],
+            'user_email': user.email,
+            'failure_url': failure_url,
+        }
+        segment.track(user.id, 'edx.bi.program.course-enrollment.nudge', event_properties)
```

Comment:
@irfanuddinahmad I have changed the name from `edx.bi.program.course-enrollment.missing` to `edx.bi.program.course-enrollment.nudge` please review.